### DISCUSSION TITLE
Screen data is not properly transmitted when using the sample client.

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8095] [Q&A] [answered]
hello.

I built the sample client with the options below and got the executable and library files.

The sample client referred to the existing client and made some customizations, and the connection was successful.

However, the problem is that the data from `gdi->primary_buffer` does not come properly from `tf_end_paint`.

All that data comes to 0.

So if convert that data into an image, it's all the same bytes of meaningless images.

And even if there is no input, the screen should be updated whenever the time of the connected rdp server changes, but the data does not come after the connection.

Is there anything I need to set to get screen data from sampleClient-endpaint?

I've changed the format setting below and fixed several parts, but it's not easy to solve.

```
	settings->OsMajorType = OSMAJORTYPE_UNIX;
	settings->OsMinorType = OSMINORTYPE_NATIVE_XSERVER;
```

Below are the build options.

`cmake -GNinja -DCHANNEL_URBDRC=ON  -DWITH_CUPS=ON -DWITH_PULSE=ON -DWITH_FAAC=ON -DWITH_FAAD2=ON -DWITH_GSM=ON -DCHANNEL_URBDRC=OFF -DWITH_CLIENT_INTERFACE=ON  -DWITH_CLIENT_SHARED=ON -DCMAKE_INSTALL_PREFIX=/usr/local/sample/result WITH_XKBFILE=OFF -DWITH_XEXT=ON -DWITH_X11=OFF -DWITH_SAMPLE=ON -DWITH_SERVER=OFF -B -H`

I'm sorry for posting questions often. Thanks to your reply, I getting a lot of help. Thank you.

**Comment by mfleisz** [comment, marked as answer]
First off - check your build options carefully (I see `-DCHANNEL_URBDRC=ON` and `-DCHANNEL_URBDRC=OFF`).

Second it is hard to help you in any way without knowing if there are any error outputs or what changes you have made to the code (i.e. why did you change the OS type? Messing with internals may cause unexpected behavior).

Also have you actually tried running the client in a debugger to see what might be wrong?

**Reply by kimcoldwater** [reply]
sorry.

The problem was that the screen did not come out because the `size` or `gdi:sw` was not specified.

I thought the this has a default value like the existing client.

After listening to your advice,

I further analyzed the CMakeList and build options of FreeRDP.

As a result, Linux Headless client creation is progressing smoothly using a sample client.

Thank you for your kind advice.

**Reply by mfleisz** [reply]
Awesome - glad to hear it's working now! :-)

**Reply by Saddingt0n** [reply]
Hi @kimcoldwater. I’m also attempting to create s headless client and I have been stuck in some places. I would appreciate it if you could share your contact details so I may trouble you with a few questions.
